src/Functions/splitByChar.cpp

```diff
@@ -66,6 +67,21 @@ class SplitByCharImpl
 
         max_substrings_includes_remaining_string = max_substrings_includes_remaining_string_;
         max_splits = extractMaxSplits(arguments, 2);
+
+
+        const ColumnString * col_str = checkAndGetColumn<ColumnString>(arguments[strings_argument_position].column.get());
+        if (col_str)
```

Comment:
No. There is another possibility that the input column is ColumnConst. I ignore it because there is no need to get reserve size under such condition.

Reply:
Okay, but then please add that as a code comment, otherwise future readers might be confused.